### PR TITLE
FEXCore: Implement AVX reconstruction helpers

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Due to the distinct view of AVX registers changing depending on if the host supports SVE256 or not, introduce some helpers to reconverge the view in the ways that Linux and Wow64 wants.

The TestHarnessRunner has a bit of a kludge but rewriting it to not read the InternalThreadState object directly will take longer than currently desired. So fetch the register state, then reconverge the AVX register state. Once the InternalThreadState object changes its view to cut out those excess 256bytes then this will get refactored at that point.

### CODE REVIEW

File: FEXCore/include/FEXCore/Core/Context.h
Comment (alyssarosenzweig) [inline FEXCore/include/FEXCore/Core/Context.h:197]: Does it make sense to drop this enum, and instead infer the recovery mode inside the helpers by checking if `YMM_High != nullptr`? This would simplify the helper signatures a fair bit and remove a failure mode (passing a mode that doesn't match whether you passed a YMM_High pointer)

**Review by alyssarosenzweig — CHANGES_REQUESTED** [review]
.

File: FEXCore/include/FEXCore/Core/Context.h
Comment (Sonicadvance1) [reply FEXCore/include/FEXCore/Core/Context.h:197]: Indeed, that might make sense

Comment (alyssarosenzweig) [reply FEXCore/include/FEXCore/Core/Context.h]: I've gone ahead and made this change, IMO it's a lot nicer now, see diff https://rosenzweig.io/diff2.txt ... approving, I'm good with this to go in if you're ok with that diff.

**Review by alyssarosenzweig — APPROVED** [review]

**Review by Sonicadvance1 — COMMENTED** [review]
I can't approve my own PR but the changes look good

**Comment by alyssarosenzweig** [comment]
> I can't approve my own PR but the changes look good

You can't approve it but you can merge it 😄

**Pull request merged**
